portfolio-project-matching-app/Firebase/clientApp.ts

```diff
@@ -19,22 +37,63 @@ const firebaseApp = initializeApp(firebaseConfig);
 const db = getFirestore();
 // const analytics = getAnalytics(firebaseApp);
 
-const getDocSnapshotById = async(collectionName, docId) => {
+/*
+  CREATE
+*/
+
+const addNewDoc = async (coll, payload) => {
   /*
-  DESCRIPTION:  gets snapshot of document from provided collection name with
-                provided document ID
+    DESCRIPTION:  adds a new document with provided payload to the specified
+                  collection. This function does not perform any input
+                  validation - that is currently left to the calling function.
+                  Note, document ID is generated by Firebase
 
-  INPUT:        string indicating desired collection name and string indicating
-                desired document ID
+    INPUT:        string indicating desired collection name and payload to be
+                  added. E.g.,
 
-  RETURN:       snapshot of document with provided document ID
+                  addNewDoc('technologies', {name: _, language: _, ...})
+
+    RETURN:       reference to new document
   */
-  const docRef = doc(db, collectionName, docId);
-  const querySnapshot = await getDoc(docRef);
-  return querySnapshot;
+  const collRef = collection(db, coll);
```

Comment:
Tried my best to go through and make my variable names a little more consistent. I could still do a ton of work in this regard. I was really learning Firebase as I went and didn't necessarily understand the nuance of references / snapshots / etc.